node/node_handler.go

```diff
@@ -559,46 +554,30 @@ func (node *Node) pingMessageHandler(msgPayload []byte, sender libp2p_peer.ID) i
 		_, ok := node.duplicatedPing.LoadOrStore(senderStr, true)
 		if ok {
 			// duplicated ping message return
-			return 0
 		}
 	}
 
-	// add to incoming peer list
-	//node.host.AddIncomingPeer(*peer)
-	node.host.ConnectHostPeer(*peer)
+	if err := node.host.ConnectHostPeer(*peer); err != nil {
+		utils.Logger().Info().Err(err).
+			Str("peer", peer.String()).
+			Msg("could not direct connect to this peer")
+	}
 
 	if ping.Node.Role != proto_node.ClientRole {
 		node.AddPeers([]*p2p.Peer{peer})
 		utils.Logger().Info().
 			Str("Peer", peer.String()).
-			Int("# Peers", node.numPeers).
+			Int("# Peers", node.host.GetPeerCount()).
 			Msg("Add Peer to Node")
 	}
-
-	return 1
 }
 
 // bootstrapConsensus is the a goroutine to check number of peers and start the consensus
 func (node *Node) bootstrapConsensus() {
 	tick := time.NewTicker(5 * time.Second)
 	defer tick.Stop()
-	lastPeerNum := node.numPeers
 	for range tick.C {
-		numPeersNow := node.numPeers
-		// no peers, wait for another tick
-		if numPeersNow == 0 {
-			utils.Logger().Info().
-				Int("numPeersNow", numPeersNow).
-				Msg("No peers, continue")
-			continue
-		} else if numPeersNow > lastPeerNum {
-			utils.Logger().Info().
-				Int("previousNumPeers", lastPeerNum).
-				Int("numPeersNow", numPeersNow).
-				Int("targetNumPeers", node.Consensus.MinPeers).
```

Comment:
This log is what we look for to see whether enough peer is obtained, which is critical in the bootstrap process. Usually when bootstrap fail. we need to double whether for this log whether it's num peer problem.

Reply:
I see, will add a log back but is the it incremental increase in peers that you want or the just if enough peers added or not?

Reply:
added a log